atlas-db-controller/databaseSchemaResource.go

```diff
@@ -146,7 +148,15 @@ func (c *Controller) syncSchema(key string) error {
 
 	ver, dirt, err := mgrt.Version()
 	if err != nil {
-		if err == migrate.ErrNilVersion {
+		errString := err.Error()
+		if strings.Contains(errString, BAD_CONN) || strings.Contains(errString, CONN_CLOSED) {
```

Comment:
This cleanup of cache is fine.
What about cleaning up during resource deletion?

Reply:
that will happen in `delete` method. This cache is global right ? there it can be accessed.

Reply:
that will happen in delete method. This cache is global right ? there it can be accessed.

But what will happen if a deletion happened already in a separate routine? will we get it as error here ?